### PR TITLE
Fix documentation bugs

### PR DESCRIPTION
**Pull request by zzibo**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -95,7 +100,14 @@ If you already know what to search for, use `CTRL-F` for Windows/Linux or `COMMA
 
 <box type="important" seamless>
 
-**IMPORTANT:** Feel free to use this table and navigate to the features that interests you by clicking on the action words 😁 
+**IMPORTANT:**
+
+**Delete and clear operations in KnottyPlanners:**
+
+All delete and clear operations (except `tag-del`) in KnottyPlanner will be followed by a confirmation prompt to avoid accidental deletion. Therefore, delete and clear operations must be followed by either `y` command (confirm delete operation) or `n` command (remove pending operation). Otherwise, following delete and clear commands may be affected. Hence, if you accidentally mistype or enter an unintended command before confirming, we suggest you to use `n` to cancel any pending delete operations to ensure everything runs smoothly 😊.
```
Comment (ywllowsensor) [inline docs/UserGuide.md:107]: "KnottyPlanners"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -278,15 +290,22 @@ Format: `edit n/NAME [new/NEW_NAME] [p/NEW_PHONE] [e/NEW_EMAIL] [a/NEW_ADDRESS]
 
 * You have to provide at least one of the optional fields.
 * Tags can't be edited, so if you'd like to change a tag, simply delete the existing one using
-['tag-del'](#deleting-tags-from-a-contact) and add a new one using ['tag-add'](#adding-tags-to-a-contact)!
+['tag-del'](#deleting-tags-from-a-contact-tag-del--td) and add a new one using ['tag-add'](#adding-tags-to-a-contact-tag-add--ta)!
```
Comment (ywllowsensor) [inline docs/UserGuide.md:293]: link is broken, shd be "....tag-del-td" and "....tag-add-ta". only one dash

File: docs/UserGuide.md [hunk 3]
```diff
@@ -345,8 +372,7 @@ Format: `del-wed w/NAME & NAME` / `dw w/NAME & NAME` followed by `y` or `n`
 
 **IMPORTANT:** 
 
-* `dw w/NAME & NAME` MUST BE followed by either of the two commands, otherwise, following delete commands may be affected.
-* If you accidentally mistype or enter an unintended command before confirming, we suggest you to use `n` to cancel any pending delete operations to ensure everything runs smoothly 😊
+* `dw w/NAME & NAME` MUST BE followed by either of the two commands, otherwise, following delete commands may be affected. Refer to [Command Summary](#command-summary) for more information ☺️.
 * The wedding must be entered exactly as it was saved in the wedding book to successfully delete it.
 * As we do not have an edit wedding command, please use ['del-wed'](#deleting-a-wedding-del-wed--dw) and ['add-wed'](#adding-a-wedding-add-wed--aw) to change details of a wedding.
```
Comment (ywllowsensor) [inline docs/UserGuide.md:377]: links broken here tooo. shd be "wed-dw" and "wed-aw"

**Review by ywllowsensor — COMMENTED** [review]
some minor nits, typos, broken links

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ Another *Sequence Diagram* below shows how the components interact with each oth
 The four main components (also shown in the diagram above, staticContext is not a main component),
 
 * defines its *API* in an `interface` with the same name as the Component.
-* implements its functionality using a concrete `{Component Name}Manager` class (which follows the corresponding API `interface` mentioned in the previous point.
+* implements its functionality using a concrete `{Component Name}Manager` class (which follows the corresponding API `interface` mentioned in the previous point).
```
Comment (jonushzw) [inline docs/DeveloperGuide.md:66]: nice

**Review by jonushzw — COMMENTED** [review]
Good work! I am very impressed

-Team Lead

**Comment by jonushzw** [comment]
LGTM!

**Pull request merged**